src/app/pages/all-meals/meal-details/meal-details.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { switchMap } from 'rxjs';
+import { MealDetails } from 'src/app/models/meal-details.model';
+import { MenuService } from 'src/app/services/menu.service';
+
+@Component({
+  selector: 'app-meal-details',
+  templateUrl: './meal-details.component.html',
+  styleUrls: ['./meal-details.component.scss']
+})
+export class MealDetailsComponent implements OnInit {
+  id: string;
+  meal: MealDetails;
+  ingredients: string[] = [];
+
+  constructor(private route: ActivatedRoute, private router: Router, private menuService: MenuService) { }
+
+  ngOnInit(): void {
+    this.route.params.pipe(
+      switchMap(params => {
+        this.id = params['id'];
+        return this.menuService.getFoodDetailsById(this.id);
+      })
+    ).subscribe(
+      (meal: any) => {
+        this.meal = meal.meals[0];        
```

Comment:
it would also be a good practice to unpack this in service, that way you'll reduce the need to write these lines in every component that displays meal